From 5d9c0179b7006c4511abb1fa3df5e37b9ce487de Mon Sep 17 00:00:00 2001
From: Geoff Levand <geoff@infradead.org>
Date: Mon, 18 May 2020 16:48:37 -0700
Subject: [PATCH 05/22] Kconfig: Add CONFIG_ARM64_ILP32

Signed-off-by: Geoff Levand <geoff@infradead.org>
---
 arch/arm64/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 5d513f461957..653e7978299d 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -1361,6 +1361,15 @@ endif
 
 endif
 
+config ARM64_ILP32
+	bool "Kernel support for ILP32"
+	select COMPAT
+	default y
+	help
+	  This option enables support for AArch64 ILP32 user space.  ILP32
+	  is an ABI where long and pointers are 32bits but it uses the AARCH64
+	  instruction set.
+
 menu "ARMv8.1 architectural features"
 
 config ARM64_HW_AFDBM
-- 
2.20.1

