From 55668b2b8f281bbd28a4964f5eb6301c4d77dd16 Mon Sep 17 00:00:00 2001
From: Geoff Levand <geoff@infradead.org>
Date: Fri, 27 Sep 2019 10:58:58 -0700
Subject: [PATCH 12/14] ptrace: Mods 1

Signed-off-by: Geoff Levand <geoff@infradead.org>
---
 arch/arm64/kernel/ptrace.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/kernel/ptrace.c b/arch/arm64/kernel/ptrace.c
index da2441d7b066..4a91d89bbc90 100644
--- a/arch/arm64/kernel/ptrace.c
+++ b/arch/arm64/kernel/ptrace.c
@@ -1685,6 +1685,10 @@ long compat_arch_ptrace(struct task_struct *child, compat_long_t request,
 	void __user *datap = compat_ptr(data);
 	int ret;
 
+	if (is_ilp32_task()) {
+		return compat_ptrace_request(child, request, caddr, cdata);
+	}
+
 	switch (request) {
 		case PTRACE_PEEKUSR:
 			ret = compat_ptrace_read_user(child, addr, datap);
-- 
2.17.1

