From af42390f3b2cb1c7f4474dec2aa3d72887cd262d Mon Sep 17 00:00:00 2001
From: Geoff Levand <geoff@infradead.org>
Date: Wed, 9 Oct 2019 17:04:02 -0700
Subject: [PATCH 05/22] Kconfig: Add CONFIG_ARM64_ILP32

Signed-off-by: Geoff Levand <geoff@infradead.org>
---
 arch/arm64/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 0b30e884e088..c6741589b48f 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -1357,6 +1357,15 @@ endif
 
 endif
 
+config ARM64_ILP32
+	bool "Kernel support for ILP32"
+	select COMPAT
+	default y
+	help
+	  This option enables support for AArch64 ILP32 user space.  ILP32
+	  is an ABI where long and pointers are 32bits but it uses the AARCH64
+	  instruction set.
+
 menu "ARMv8.1 architectural features"
 
 config ARM64_HW_AFDBM
-- 
2.20.1

