From 4de7b9ed0b0a472cf7ad21691249ff010aa24f1b Mon Sep 17 00:00:00 2001
From: Geoff Levand <geoff@infradead.org>
Date: Fri, 31 Jan 2020 15:14:00 -0800
Subject: [PATCH 19/22] Work-around for ILP32 kernel oops in elf_core_dump

Signed-off-by: Geoff Levand <geoff@infradead.org>
---
 fs/coredump.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/coredump.c b/fs/coredump.c
index f8296a82d01d..234ae352d1b1 100644
--- a/fs/coredump.c
+++ b/fs/coredump.c
@@ -594,7 +594,7 @@ void do_coredump(const kernel_siginfo_t *siginfo)
 	audit_core_dumps(siginfo->si_signo);
 
 	binfmt = mm->binfmt;
-	if (!binfmt || !binfmt->core_dump)
+	if (!binfmt || !binfmt->core_dump || test_thread_flag(TIF_ILP32))
 		goto fail;
 	if (!__get_dumpable(cprm.mm_flags))
 		goto fail;
-- 
2.20.1

